webrender/src/texture_cache.rs

```diff
@@ -327,18 +327,29 @@ impl TexturePage {
         self.dirty = true
     }
 
-    fn grow(&mut self, new_texture_size: u32) {
-        self.free_list.push(&DeviceUintRect::new(
-            DeviceUintPoint::new(self.texture_size, 0),
-            DeviceUintSize::new(new_texture_size - self.texture_size, new_texture_size)));
-        self.free_list.push(&DeviceUintRect::new(
-            DeviceUintPoint::new(0, self.texture_size),
-            DeviceUintSize::new(self.texture_size, new_texture_size - self.texture_size)));
+    fn grow(&mut self, new_texture_size: DeviceUintSize) {
+        let new_rects = [
```

Comment:
we should probably assert on `new_texture_size >= texture_size` here, since the code expects it to be true and would do strange things in release (while failing on arithmetic overflow in debug)